src/Disks/VolumeJBOD.cpp

```diff
@@ -101,6 +102,11 @@ DiskPtr VolumeJBOD::getDisk(size_t /* index */) const
         case VolumeLoadBalancing::LEAST_USED:
         {
             std::lock_guard lock(mutex);
+            if (!least_used_ttl_ms || least_used_update_watch.elapsedMilliseconds() >= least_used_ttl_ms)
+            {
+                disks_by_size = LeastUsedDisksQueue(disks.begin(), disks.end());
+                least_used_update_watch.restart();
```

Comment:
Feel like a small repetition of the same logic and below. Let's extract it into a method.

Reply:
The problem is that it requires mutex, and TSA does not understand the it is held in the outer scope, so you can't write something like this:

```patch
bool VolumeJBOD::updateDisksSizesIfExpired(std::lock_guard<std::mutex> &) const
{
    if (!least_used_ttl_ms || least_used_update_watch.elapsedMilliseconds() >= least_used_ttl_ms)
    {
        disks_by_size = LeastUsedDisksQueue(disks.begin(), disks.end());
        least_used_update_watch.restart();
        return true;
    }
    return false;
}
````

It could be done by a) removing TSA annotation and b) re-holding the lock, but I doubt that this complexity worth it. @serxa What do you think?

Reply:
Ufff... TSA makes us do weird stuff... Okay, let's keep it is!